src/util/getRoutePath.js

```diff
@@ -1,10 +1,14 @@
-export default function getRoutePath(route, routes) {
+export default function getRoutePath(route, routes, name) {
   const matchIndex = routes.indexOf(route);
   if (matchIndex < 0) {
     throw new Error('`route` not found in `routes`');
   }
   const routesUntil = routes.slice(0, matchIndex + 1);
-  return routesUntil
+  let routePath = routesUntil
     .reduce((acc, { path }) => acc.concat(path), [])
     .join('/');
+  if (name) {
+    routePath = `${routePath}@${name}`;
```

Comment:
Is this unique enough? If I understand this correctly this might break down if we have two similar routes where one uses named components.

_Example:_
1. `about/my-path` and a named component named `hello`.
2. `about/my-path@hello`.

Maybe we can go for a symbol that is forbidden in URLs, as I understand it `@` is reserved but not fully forbidden.